docs/framework/guides/creating-simple-plugin-timestamp.md

```diff
@@ -0,0 +1,271 @@
+import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
+import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+class Timestamp extends Plugin {
+    init() {
+        console.log( 'Timestamp was initialized.' );
+    }
+}
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+
+		//Add the Timestamp plugin to config.plugins array
+		plugins: [ 
+			Timestamp, Essentials, Paragraph, Heading, List, Bold, Italic
+			], 
+		toolbar: [ 'heading', 'bold', 'italic', 'numberedList', 'bulletedList' ]
+	} )
+	.then( editor => {
+		console.log( 'Editor was initialized', editor );
+	} )
+	.catch( error => {
+		console.error( error.stack );
+	} );
+```
+Rebuild and check in your console if the timestamp was initialized. You should see this: SCREENSHOT
+
+## Registering a toolbar button
+
+CKEditor 5 has a rich UI library, from where we'll grab the {@link module:ui/button/buttonview~ButtonView `ButtonView`} class for our toolbar button. 
+
+Once we create a new instance of the `ButtonView`, we'll be able to customize it by setting its properties. We'll create a label, which will be visible on the button thanks to the `withText` property. 
+
+We need to register our button in the editor's UI {@link module:ui/componentfactory~ComponentFactory `componentFactory`}, so it can be displayed in the toolbar. We'll pass the name of the button in the `componentFactory.add` method, so we'll be able to add it into the `config.toolbar` array.
+
+```js
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+
+class Timestamp extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		// The button must be registered among the UI components of the editor
+		// to be displayed in the toolbar.
+		editor.ui.componentFactory.add( 'timestamp', locale => {
+			
+			// The button will be an instance of ButtonView
+			const button = new ButtonView( locale );
+
+			button.set( {
+				label: 'Timestamp',
+				withText: true
+			} );
+
+			return button;
+		} );
+	}
+}
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ 
+			Timestamp, Essentials, Paragraph, Heading, List, Bold, Italic 
+			], 
+
+		//Add the Timestamp button to the config.toolbar array
+		toolbar: [ 
+			'timestamp', 'heading', 'bold', 'italic', 'numberedList', 'bulletedList'
+			] 
+	} )
+	.then( editor => {
+		console.log( 'Editor was initialized', editor );
+	} )
+	.catch( error => {
+		console.error( error.stack );
+	} );
+
+```
+You should be able to see the timestamp button now. It doesn't do anything just yet, so let's change that. 
+
+## Inserting a timestamp
+
+We can now define the core funcitonality of our plugin, the action that should be executed once our button is clicked. 
+
+When we want to insert something into the document structure (or change it in any other way), we need to use the model writer, available in the model's `change()` method. 
```

Comment:
Link into to model guide or model writer API or writer change method. Or better yet, to this section `framework/guides/architecture/editing-engine.html#changing-the-model`.